packages/flutter_tools/test/integration.shard/flutter_build_windows_test.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/io.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+void main() {
+  // Only run this test on Windows
+  if (!platform.isWindows) {
+    return;
+  }
+
+  late Directory tempDir;
+  late Directory projectRoot;
+  late String flutterBin;
+  late Directory releaseDir;
+  late File exeFile;
+
+  setUpAll(() {
+    tempDir = createResolvedTempDirectorySync('build_windows_test.');
+    flutterBin = fileSystem.path.join(
+      getFlutterRoot(),
+      'bin',
+      'flutter',
+    );
+    processManager.runSync(<String>[flutterBin, 'config',
+      '--enable-windows-desktop',
+    ]);
+
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      'hello',
+    ], workingDirectory: tempDir.path);
+
+    projectRoot = tempDir.childDirectory('hello');
+
+    releaseDir = fileSystem.directory(fileSystem.path.join(
+      projectRoot.path,
+      'build',
+      'windows',
+      'runner',
+      'Release'
+    ));
+
+    exeFile = fileSystem.file(fileSystem.path.join(
+      releaseDir.path,
+      'hello.exe'
+    ));
+  });
+
+  tearDownAll(() {
+    tryToDelete(tempDir);
+  });
+
+  testWithoutContext('flutter build windows creates exe', () {
+    final ProcessResult result = processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'build',
+      'windows',
+      '--no-pub',
+    ], workingDirectory: projectRoot.path);
+
+    expect(result.exitCode, 0);
+    expect(releaseDir, exists);
+    expect(exeFile, exists);
+
+    // Default exe has version 1.0.0
+    final String fileVersion = _getFileVersion(exeFile);
+    final String productVersion = _getProductVersion(exeFile);
+
+    expect(fileVersion, equals('1.0.0.0'));
```

Comment:
This is not ideal. Since Flutter's builder number defaults to `1`, the file version should be `1.0.0.1`. I'll fix this in a subsequent PR.